process_data.sh

```diff
@@ -0,0 +1,309 @@
+  FILESEG="${file}_seg"
+  FILESEGMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/${folder_contrast}/${FILESEG}-manual.nii.gz"
+  echo
+  echo "Looking for manual segmentation: $FILESEGMANUAL"
+  if [[ -e $FILESEGMANUAL ]]; then
+    echo "Found! Using manual segmentation."
+    rsync -avzh $FILESEGMANUAL ${FILESEG}.nii.gz
+    sct_qc -i ${file}.nii.gz -s ${FILESEG}.nii.gz -p sct_deepseg_sc -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  else
+    echo "Not found. Proceeding with automatic segmentation."
+    # Segment spinal cord
+    sct_deepseg_sc -i ${file}.nii.gz -c $contrast -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  fi
+}
+
+
+# T1w
+# ------------------------------------------------------------------------------
+file_t1="${SUBJECT}_T1w"
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist $file_t1 "t1"
+file_t1_seg=$FILESEG
+
+# T2w
+# ------------------------------------------------------------------------------
+file_t2="${SUBJECT}_T2w"
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist $file_t2 "t2"
+file_t2_seg=$FILESEG
+
+# T2s
+# ------------------------------------------------------------------------------
+file_t2s="${SUBJECT}_T2star"
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist $file_t2s "t2s" 
+file_t2s_seg=$FILESEG
+
+# MTS
+# ------------------------------------------------------------------------------
+file_t1w="${SUBJECT}_acq-T1w_MTS"
+file_mton="${SUBJECT}_acq-MTon_MTS"
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist $file_t1w "t1" 
+file_t1w_seg=$FILESEG
+segment_if_does_not_exist $file_mton "t2s"
+file_mton_seg=$FILESEG
+
+
+# DWI
+# ------------------------------------------------------------------------------
+cd ../dwi
+file_dwi_mean="${SUBJECT}_rec-average_dwi"
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist ${file_dwi_mean} "dwi"
+file_dwi_mean_seg=$FILESEG
+
+
+# Go back to parent folder
+cd ../anat
+
+# Registration
+# ------------------------------------------------------------------------------
+# Create mask
+file_t2_mask="${file_t2_seg}_mask"
+sct_create_mask -i ${file_t2}.nii.gz -p centerline,${file_t2_seg}.nii.gz -size 55mm -o ${file_t2_mask}.nii.gz 
+
+# Register T1w to T2w
+# With type=im
+#sct_register_multimodal -i ${file_t1}.nii.gz -d ${file_t2}.nii.gz -iseg ${file_t1_seg}.nii.gz -dseg ${file_t2_seg}.nii.gz -m ${file_t2_mask}.nii.gz -param step=1,type=im,algo=slicereg,metric=CC,iter=10,shrink=4,poly=2:step=2,type=im,algo=slicereg,metric=CC,iter=10,shrink=2 -qc ${PATH_QC} -qc-subject ${SUBJECT} 
+# With type=seg : BEST 
+sct_register_multimodal -i ${file_t1}.nii.gz -d ${file_t2}.nii.gz -iseg ${file_t1_seg}.nii.gz -dseg ${file_t2_seg}.nii.gz -param step=1,type=seg,algo=slicereg,metric=CC,iter=10,shrink=4,poly=2:step=2,type=seg,algo=slicereg,metric=CC,iter=10,shrink=2 -qc ${PATH_QC} -qc-subject ${SUBJECT} 
```

Comment:
- shrink factor does not apply to slicereg
- in general, better to use MeanSquare with segmentation-based registration
- no need for two steps with slicereg (because shrink factor not accounted for)

```suggestion
sct_register_multimodal -i ${file_t1}.nii.gz -d ${file_t2}.nii.gz -iseg ${file_t1_seg}.nii.gz -dseg ${file_t2_seg}.nii.gz -param step=1,type=seg,algo=slicereg,metric=MeanSquares,iter=10,poly=2 -qc ${PATH_QC} -qc-subject ${SUBJECT} 
```

Reply:
alternative:
- sct_register_multimodal -i ${file_t1}.nii.gz -d ${file_t2}.nii.gz -iseg ${file_t1_seg}.nii.gz -dseg ${file_t2_seg}.nii.gz -m ${file_t2_mask}.nii.gz -param step=1,type=seg,algo=slicereg,metric=MeanSquares,iter=10,poly=2:step=2,type=im,algo=slicereg,metric=CC,iter=10,poly=2 -qc ${PATH_QC} -qc-subject ${SUBJECT}